src/wp-includes/media.php

```diff
@@ -3430,7 +3432,20 @@ function wp_audio_shortcode( $attr, $content = '' ) {
 	$attr_strings = array();
 
 	foreach ( $html_atts as $k => $v ) {
-		$attr_strings[] = $k . '="' . esc_attr( $v ) . '"';
+		if ( in_array( $k, array( 'loop', 'autoplay', 'muted' ), true ) && true === $v ) {
+			// Add boolean attributes without a value
+			$attr_strings[] = esc_attr( $k );
+		} elseif ( 'preload' === $k && ! empty( $v ) ) {
+			// Handle the preload attribute with specific allowed values
+			$allowed_preload_values = array( 'none', 'metadata', 'auto' );
+			if ( in_array( $v, $allowed_preload_values, true ) ) {
+				$attr_strings[] = sprintf( '%s="%s"', esc_attr( $k ), esc_attr( $v ) );
+			}
+			// If the value is not allowed, you can log a warning or handle it as needed
+		} else {
+			// For other attributes, include the value
+			$attr_strings[] = $k . '="' . esc_attr( $v ) . '"';
+		}
 	}
```

Comment:
for improved readability, i suggest to switch to better readable variable.

`foreach ( $html_atts as $attribute => $value ) {` will reduce cognitive load, imho `$k` and `$v` were ok for the oneliner, but in this case I would alter to longer and self explanatory variables.

I would add a suggestion, but it would miss out on line 3432.